src/Microsoft.DotNet.Cli.Sln.Internal/SlnFile.cs

```diff
@@ -698,17 +698,34 @@ internal void ReadLine(string line, int currentLine)
 
         internal void Write(TextWriter writer)
         {
-            foreach (DictionaryEntry e in _values)
+            if (_isMetadata)
             {
-                if (!_isMetadata)
+                foreach (DictionaryEntry e in _values)
                 {
-                    writer.Write("\t\t");
+                    if (Id != null)
+                    {
+                        writer.Write(Id + ".");
+                    }
+                    writer.WriteLine(e.Key + " = " + e.Value);
```

Comment:
I think if you give part of sln file which is being generated here or name a dictionary so that it's obvious which part it is that should be enough (i.e. applesToOranges - this way you know that apples map to oranges; key-value relationship is too generic).